Student/Student/Routes.swift

```diff
@@ -221,18 +222,48 @@ let routeMap: KeyValuePairs<String, RouteHandler.ViewFactory?> = [
         return ModuleListViewController.create(courseID: courseID, moduleID: moduleID)
     },
 
-    "/courses/:courseID/modules/items/:itemID": { _, params in
+    "/courses/:courseID/modules/items/:itemID": { url, params in
         guard let courseID = params["courseID"], let itemID = params["itemID"] else { return nil }
         guard let session = Session.current else { return nil }
+        if ExperimentalFeature.studentModules.isEnabled {
+            return ModuleItemSequenceViewController.create(
+                courseID: courseID,
+                assetType: .moduleItem,
+                assetID: itemID,
+                url: url
+            )
+        }
         return try? ModuleItemDetailViewController(session: session, courseID: courseID, moduleItemID: itemID, route: route)
     },
 
-    "/courses/:courseID/modules/:moduleID/items/:itemID": { _, params in
+    "/courses/:courseID/modules/:moduleID/items/:itemID": { url, params in
         guard let courseID = params["courseID"], let itemID = params["itemID"] else { return nil }
         guard let session = Session.current else { return nil }
+        if ExperimentalFeature.studentModules.isEnabled {
+            return ModuleItemSequenceViewController.create(
+                courseID: courseID,
+                assetType: .moduleItem,
+                assetID: itemID,
+                url: url
+            )
+        }
         return try? ModuleItemDetailViewController(session: session, courseID: courseID, moduleItemID: itemID, route: route)
     },
 
+    "/courses/:courseID/module_item_redirect/:itemID": { url, params in
```

Comment:
is this a route we made up, or do we get urls with this in it from the api?

Reply:
It is a canvas route. It is the `html_url` of External URL module items.